### PR TITLE
Added _parse_on_conflict 

### PR DESCRIPTION
**Pull request by cheunhong**
Added parsing support for `INSERT ON CONFLICT` statement. 
As this is my first contribution appreaciate any feedback on improvement needed

### CODE REVIEW

**Comment by tobymao** [comment]
please run the linters and tests

make check

File: sqlglot/tokens.py [hunk 1]
```diff
@@ -156,6 +156,7 @@ class TokenType(AutoName):
     COMMIT = auto()
     COMPOUND = auto()
     CONSTRAINT = auto()
+    CONFLICT = auto()
```
Comment (tobymao) [inline sqlglot/tokens.py:159]: delete all of these

File: sqlglot/parser.py [hunk 2]
```diff
@@ -1601,11 +1601,40 @@ def _parse_insert(self) -> exp.Expression:
...
             partition=self._parse_partition(),
             expression=self._parse_ddl_select(),
+            on_conflict=self._parse_on_conflict(),
             returning=self._parse_returning(),
             overwrite=overwrite,
             alternative=alternative,
         )
 
+    def _parse_on_conflict(self) -> t.Optional[exp.Expression]:
+        if not self._match(TokenType.ON):
```
Comment (tobymao) [inline sqlglot/parser.py:1611]: this should be

conflict = self._match_seq("ON", "CONFLICT")
duplicate = self._match_seq("ON", "DUPLICATE KEY")

if not conflict or duplicate:
   return None

File: sqlglot/parser.py [hunk 3]
```diff
@@ -1601,11 +1601,40 @@ def _parse_insert(self) -> exp.Expression:
...
+        nothing = False
+        expressions = None
+        key = None
+        constraint = None
+
+        if self._match(TokenType.CONFLICT):
+            key = self._parse_csv(self._parse_value)
+            if self._match(TokenType.ON):
+                self._match(TokenType.CONSTRAINT)
+                constraint_name = self._parse_var()
```
Comment (tobymao) [inline sqlglot/parser.py:1623]: why not self._parse_id_var?

File: sqlglot/parser.py [hunk 4]
```diff
@@ -1601,11 +1601,40 @@ def _parse_insert(self) -> exp.Expression:
...
+                if constraint_name:
+                    constraint = exp.Identifier(this=constraint_name.this)
+            self._match(TokenType.DO)
+            if self._match(TokenType.NOTHING):
```
Comment (tobymao) [inline sqlglot/parser.py:1627]: nothing = self._match_seq("nothing")

**Comment by tobymao** [comment]
you need to add tests and a generator

**Comment by tobymao** [comment]
thanks for the PR! i've left a couple of comments

**Comment by cheunhong** [comment]
@tobymao have added the changes accordingly

**Pull request merged**